addons/nodot/characters/Fly3D.gd

```diff
@@ -0,0 +1,143 @@
+## This node is a mechanic for fly player (experimental)
+class_name Fly3D extends Node3D
+
+## This signal call in "_physics_process()" of player: $(Fly3D).emit_signal("config_fly", self, $(model of player))
+signal config_fly(Parent, model)
+
+@export_category("Configs Funcs")
+## Can rotate model of player
+@export var can_rotate = true
+## when scene start, move is automatic
+@export var automatic_move = false
+## Check when it can start to fly
+@export var limit_high = 2.0
+
+
+@export_category("Velocity Configs")
+## always speed
+@export var default_speed : int = 2
+## min of speed
+@export var min_speed : int = 1
+## max of speed
+@export var max_speed : int = 5
+
+@export_category("Rotation Action")
+## Frame x second to rotate model
+@export var time_to_rot = 1.0
+## Grades for rotate
+@export var rotate_grade = 1.0
+## Conditional for rotate model (grades)
+@export_range(0, 180) var limit_rotation : int = 90
+
+## Local velocity
+var velocity = Vector3()
+## Temp space from default speed
+var temp_default_speed
+
+func _enter_tree():
+	temp_default_speed = default_speed
+	
+	if get_parent() is CharacterBody3D:
+		connect("config_fly", _fly)
+	else:
+		print_debug("Parent must have CharacterBody3D, no: ", get_parent())
+
+func _fly(Parent, model):
+	
+	## Check if can start to fly
+	if !Parent.is_on_floor() and Parent.position.y >= limit_high:
+		
+		get_input(model)
+		
+		## send local velocity
+		Parent.set_velocity(velocity)
+		## local move and slide
+		Parent.move_and_slide()
+	else:
+		return
+	
+func get_input(model):
+	## Config new controlls
+	var input_dir = Input.get_vector("ui_fly_left", "ui_fly_right", "ui_fly_foward", "ui_fly_back")
```

Comment:
Check the other nodes for action naming conventions and setup. These actions need to be automatically defined with default controls for both keyboard and controller.